usr/share/rear/layout/save/GNU/Linux/210_raid_layout.sh

```diff
@@ -207,6 +278,8 @@ mdadm --detail --scan --config=partitions | while read array raiddevice junk ; d
     # will create a 'mdadm' command in diskrestore.sh like "mdadm ... --layout=-unknown- ..." which would fail
     # during "rear recover" with something like "mdadm: layout -unknown- not understood for raid0"
     # see https://github.com/rear/rear/issues/2616
+    # and ensure $layout is a single non empty and non blank word
+    # (no quoting because test " " returns zero exit code):
```

Comment:
But if $layout is empty or blank, the command will become
```
test -a '-unknown-' != ""
```
I don't think that's intended. See also http://www.oilshell.org/blog/2017/08/31.html

(OTOH, you don't need to solve all existing bugs with this cleanup)

Reply:
Thank you for spotting it!
It works but not cleanly.
I will use
```
test $layout -a '-unknown-' != "$layout" && raid_layout_entry+=" layout=$layout
```

What I get on command line:
```
# unset layout

# ( set -x ; test $layout -a '-unknown-' != "$layout" && echo Y )
+ test -a -unknown- '!=' ''
bash: test: too many arguments

# layout="-unknown-"
# ( set -x ; test $layout && test '-unknown-' != "$layout" && echo Y )
+ test -unknown-
+ test -unknown- '!=' -unknown-

# layout="n2"
# ( set -x ; test $layout && test '-unknown-' != "$layout" && echo Y )
+ test n2
+ test -unknown- '!=' n2
+ echo Y
Y
```

Reply:
Should be fixed via
https://github.com/rear/rear/pull/2768/commits/5d5dfedda3f4d36546618288515428b550543344

I won't try to solve all existing bugs in this script with this cleanup
but I intend to have the RAID10 layout support code cleaned up.